BambeeHR/pollen
# PRIVA-1436 - UserSelectInput - From SelectInput => UserSelectInput

## Issue(s)
Create a copy of SelectInput and update it to accommodate for new UserSelectInput design and functionalities. 

## Description
Takes the SelectInput.

1. add and use #option slot for UL/LI Options List
2. Style and override vue-select css with new style per design
![Screenshot_20220802_124752](https://user-images.githubusercontent.com/10406785/182460643-5989024b-4612-44da-9a53-47fa137c0213.png)

3. Include Search and MultiSelect functionalities.

## Screenshot(s)
Options Opened.
![Screenshot_20220802_125036](https://user-images.githubusercontent.com/10406785/182461071-13cf4612-54e1-4e28-9945-f98f61ca720c.png)

Hover over an option to see the purple tinted highlight
![Screenshot_20220802_125057](https://user-images.githubusercontent.com/10406785/182461132-1d1c1a1e-18e3-4c20-8221-711d184ef208.png)

Selected State - Single
![Screenshot_20220802_125121](https://user-images.githubusercontent.com/10406785/182461191-cac9ae4a-9006-4756-ae79-04b1b821d9f4.png)

Multi Selected
![Screenshot_20220802_125202](https://user-images.githubusercontent.com/10406785/182461293-7634b95b-8a71-42db-8c26-d16dc05f31ef.png)

Search Feature
![Screenshot_20220802_130058](https://user-images.githubusercontent.com/10406785/182462785-b121c1ed-a571-43fa-a846-6a7efac29cbf.png)

Search Null Result
![Screenshot_20220802_130608](https://user-images.githubusercontent.com/10406785/182463586-93389c90-d166-404f-aa4b-f5693429b859.png)


## Dependencies
Most of the code is a copy of the SelectInput component.
PRIVA-1436

## Testing
Open storybook preview and navigate to UserSelectInput

1. open up the selection options and see the formatted Option List.
2. Type in directly into the input field to see search functionality intact
3. Scroll down to multi select view and attempt to select multiple option items.
4. See that the Name/label is shown as "selected" in a BaseChip component.


## Risk Assessment
LOW

Currently not being used.  New Component.
